generic3g/actions/CMakeLists.txt

```diff
@@ -8,6 +8,7 @@ target_sources(MAPL.generic3g PRIVATE
   VerticalRegridAction.F90
   CopyAction.F90
   ConvertUnitsAction.F90
+  ConvertVerticalDimSpecAction.F90
```

Comment:
I very much hope that this action is only involved when an export spec is on a different staggering than the import spec.   This case could probably even be disallowed.

I also suggest that "DimSpec" in the name is misleading here as it also allows for things like surface fields.   Something more like ConvertVerticalStaggerAction.F90 seems more correct.

Reply:
> I very much hope that this action is only involved when an export spec is on a different staggering than the import spec. This case could probably even be disallowed.

Isn't that the case with HISTORY? It's currently happening here
https://github.com/GEOS-ESM/MAPL/blob/16073616abc2ca7e7b8ad37a2e50bbb5f954e4ed/base/MAPL_VerticalMethods.F90#L207

>I also suggest that "DimSpec" in the name is misleading here as it also allows for things like surface fields. Something more like ConvertVerticalStaggerAction.F90 seems more correct.

Not sure I understand. This 'action' is converting DimSpec.

Reply:
I misspoke about the situation where the action might be required.   I don't think we need it at all.   Let me explain.

The model (FV) will provide both PLE and PL.    History will either use "native" (which will choose between PLE and PL) or has an entirely different grid (fixed levels) in which case you can interpolate from either.  (Mathematically equivalent, but PLE is preferred due to edge-of-domain issues.)

At the other end, if ExtData files only provide say PLE, then rather than have an adaptor, I would require ExtData to also include PLE in its state and precompute that from PLE.   The hard case here is when a model quantity is say on centers - you will end up with a different sparse matrix than you would for edges.     But that's it.   No need to first regrid to edges and then interpolate to centers.    I can be convinced otherwise, but I think it would add needless expense and complexity.

Vertical staggering should always just result in selecting a different field for specifying the vertical coordinates when "same source", and otherwise just directly compute the needed sparse matrix.